lib/typescript/httpclient/endpoints/listTemplates.ts

```diff
@@ -0,0 +1,15 @@
+import {ListTemplatesRequestPayload, ListTemplatesPayload} from '../payload';
+import {HttpClient} from '../client';
+/* eslint-disable @typescript-eslint/no-var-requires */
+const camelcaseKeys = require('camelcase-keys');
+
+export default HttpClient.prototype.listTemplates = async function listTemplates(
+  requestPayload: ListTemplatesRequestPayload
+) {
+  const response: ListTemplatesPayload = await this.doFetchFromBackend('templates.list', {
+    name: requestPayload.name,
+    source_type: requestPayload.sourceType,
+  });
+
+  return camelcaseKeys(response.data);
+};
```

Comment:
I know @bitboxer  made some changes to the TS http library (will rebase/edit when this is merged)